source/jormungandr/jormungandr/realtime_schedule/sytral.py

```diff
@@ -0,0 +1,183 @@
+        except pybreaker.CircuitBreakerError as e:
+            logging.getLogger(__name__).error(
+                'systralRT RT service dead, using base ' 'schedule (error: {}'.format(e),
+                extra={'rt_system_id': unicode(self.rt_system_id)},
+            )
+            raise RealtimeProxyError('circuit breaker open')
+        except requests.Timeout as t:
+            logging.getLogger(__name__).error(
+                'systralRT RT service timeout, using base ' 'schedule (error: {}'.format(t),
+                extra={'rt_system_id': unicode(self.rt_system_id)},
+            )
+            raise RealtimeProxyError('timeout')
+        except Exception as e:
+            logging.getLogger(__name__).exception(
+                'systralRT RT error, using base schedule', extra={'rt_system_id': unicode(self.rt_system_id)}
+            )
+            raise RealtimeProxyError(str(e))
+
+    def _make_url(self, route_point):
+        """
+        The url returns something like a departure on a stop point
+        """
+
+        stop_id = route_point.fetch_stop_id(self.object_id_tag)
+
+        if not stop_id:
+            # one a the id is missing, we'll not find any realtime
+            logging.getLogger(__name__).debug(
+                'missing realtime id for {obj}: stop code={s}'.format(obj=route_point, s=stop_id),
+                extra={'rt_system_id': unicode(self.rt_system_id)},
+            )
+            self.record_internal_failure('missing id')
+            return None
+
+        url = "{base_url}?stop_id={stop_id}".format(base_url=self.service_url, stop_id=stop_id)
+
+        return url
+
+    def _get_dt(self, datetime_str):
+        dt = aniso8601.parse_datetime(datetime_str)
+
+        utc_dt = dt.astimezone(pytz.utc)
+
+        return utc_dt
+
+    def _get_passages(self, route_point, resp):
+        logging.getLogger(__name__).debug(
+            'sytralrt response: {}'.format(resp), extra={'rt_system_id': unicode(self.rt_system_id)}
+        )
+
+        line_code = route_point.fetch_line_id(self.object_id_tag)
+
+        departures = resp.get('departures', [])
+        next_passages = []
+        for next_expected_st in departures:
+            if line_code != next_expected_st['line']:
+                continue
+            dt = self._get_dt(next_expected_st['datetime'])
+            direction = next_expected_st.get('direction_name')
+            is_real_time = next_expected_st.get('type') == 'E'
+            next_passage = RealTimePassage(dt, direction, is_real_time)
+            next_passages.append(next_passage)
+
+        return next_passages
+
+    def _get_next_passage_for_route_point(
+        self, route_point, count=None, from_dt=None, current_dt=None, duration=None
+    ):
+        url = self._make_url(route_point)
+        if not url:
+            return None
+        r = self._call(url)
+        if not r:
+            return None
```

Comment:
If `r` is null, something surely went wrong. Maybe you want to log this ?

Reply:
good catch, in fact `r` can't be null :)